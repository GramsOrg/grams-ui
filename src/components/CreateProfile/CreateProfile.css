.ltr {
    text-align: left;
}

.rtl {
    direction: rtl;
    text-align: right;
}