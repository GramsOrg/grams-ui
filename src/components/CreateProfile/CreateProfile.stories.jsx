import React from 'react';

import { action } from "@storybook/addon-actions";

import CreateProfile from './CreateProfile.tsx';

export default {
  title: 'Onboarding/CreateProfile',
  component: CreateProfile,
  parameters: {
    componentSubtitle: 'Displays a form for the user to create their profile',
    storyshots: { disable: false },
  },
  argTypes: {
    dir: {
      control: 'radio',
      options: ['ltr', 'rtl']
    },
    inverted: {
      control: 'boolean'
    }
  }
};

const Template = ({dir, inverted, ...props}) => {
  return (
    <CreateProfile
      dir={dir}
      inverted={inverted}
      {...props}
    />
  );
};

export const Default = Template.bind({});
Default.args = {
  dir: 'ltr',
  inverted: false
}

export const RTL = Template.bind({});
RTL.args = {
  dir: 'rtl',
  inverted: false
}
