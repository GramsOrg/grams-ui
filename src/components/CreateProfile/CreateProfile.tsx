import React, { useState } from 'react';

import { useTranslation } from 'react-i18next';

import { useDirection } from 'grams-common';

import { Accordion, Button, Form, Header, Icon, Input, Label, Message, Progress } from 'semantic-ui-react';

import './CreateProfile.css';

export interface ICreateProfileProps {

  /**
   * Toggles between light and dark modes
   */
  inverted?: boolean;

  /**
   * Enables/disables form validation
   */
  validate?: boolean;

  /**
   * The callback function to be called when the user creates a profile.
   * @param {string} name - The user's name.
   * @param {string} password - The user's password.
   * @param {boolean} isDeveloper - If the user opted for developer mode
   */
  onCreate?: (name: string, password: string, isDeveloper: boolean) => void;
}

const defaultProps = {
  inverted: false,
  validate: true
};

const CreateProfile = (props: ICreateProfileProps) => {

  const { t } = useTranslation();
  const direction = useDirection();

  const [name, setName] = useState<string>('');
  const [password, setPassword] = useState<string>('');
  const [confirmPassword, setConfirmPassword] = useState<string>('');
  const [showPassword, setShowPassword] = useState<boolean>(false);
  const [developerMode, setDeveloperMode] = useState<boolean>(false);

  const position: 'left' | undefined = direction === 'ltr' ? 'left' : undefined;

  const isValid = (): boolean => {
    return !props.validate || false;
  }

  return (
    <Form>
      <Header className='centered' as="h2">{t('CreateProfile.title')}</Header>
      <Form.Input
        fluid
        icon='user'
        iconPosition={position}
        placeholder={t('CreateProfile.name')}
        value={name}
        onChange={ (e) => setName(e.target.value) }
      />
      <Form.Input
        fluid
        icon='key'
        iconPosition={position}
        placeholder={t('CreateProfile.password')}
        value={password}
        onChange={ (e) => setPassword(e.target.value) }
      />
      <Form.Input
        fluid
        icon='key'
        iconPosition={position}
        placeholder={t('CreateProfile.confirmPassword')}
        value={confirmPassword}
        onChange={ (e) => setConfirmPassword(e.target.value) }
      />
      <Button
          fluid
          type="submit"
          primary
          disabled={!isValid()}
          onClick={() => props?.onCreate?.(name, password, developerMode)}
        >
          {t('CreateProfile.create')}
        </Button>
    </Form>
  );
};

CreateProfile.defaultProps = defaultProps;

export default CreateProfile;
