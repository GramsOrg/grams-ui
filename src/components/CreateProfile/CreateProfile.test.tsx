/**
 * @jest-environment jsdom
 */

import React from 'react';
import { describe, expect, test } from '@jest/globals';
import { render } from '@testing-library/react';

import CreateProfile from './CreateProfile';

describe('CreateProfile Component', () => {
  test('it should match the snapshot', () => {
    const { asFragment } = render(<CreateProfile />);
    expect(asFragment()).toMatchSnapshot();
  });
});
