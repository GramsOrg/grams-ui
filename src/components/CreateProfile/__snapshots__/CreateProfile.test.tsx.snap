// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateProfile Component it should match the snapshot 1`] = `
<DocumentFragment>
  <form
    class="ui form"
  >
    <h2
      class="ui header centered"
    >
      Create profile
    </h2>
    <div
      class="field"
    >
      <div
        class="ui fluid left icon input"
      >
        <input
          placeholder="Name"
          type="text"
          value=""
        />
        <i
          aria-hidden="true"
          class="user icon"
        />
      </div>
    </div>
    <div
      class="field"
    >
      <div
        class="ui fluid left icon input"
      >
        <input
          placeholder="Password"
          type="text"
          value=""
        />
        <i
          aria-hidden="true"
          class="key icon"
        />
      </div>
    </div>
    <div
      class="field"
    >
      <div
        class="ui fluid left icon input"
      >
        <input
          placeholder="Confirm password"
          type="text"
          value=""
        />
        <i
          aria-hidden="true"
          class="key icon"
        />
      </div>
    </div>
    <button
      class="ui fluid primary disabled button"
      disabled=""
      tabindex="-1"
      type="submit"
    >
      Create
    </button>
  </form>
</DocumentFragment>
`;
